sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/moduls/PrepareWrapperGIT.java

```diff
@@ -0,0 +1,200 @@
+            location = location.replace(":", "/");
+            repositoryURL = location.replace("git@", preFix);
+        } else if (location.contains("http://")) {
+            repositoryURL = location.replace("http://", preFix);
+        } else if (location.contains("git://")) {
+            repositoryURL = location.replace("git://", preFix);
+        } else {
+            repositoryURL = preFix + location;
+        }
+        return repositoryURL;
+    }
+
+    private void waitForProcessToFinish() {
+
+        /* watching */
+        String watcherThreadName = "PDSPrepareWrapper-git-watcher";
+
+        LOG.debug("Start watcher thread: {}", watcherThreadName);
+
+        StreamDataRefreshRequestWatcherRunnable streamDatawatcherRunnable = new StreamDataRefreshRequestWatcherRunnable();
+        Thread streamDataWatcherThread = new Thread(streamDatawatcherRunnable);
+        streamDataWatcherThread.setName(watcherThreadName);
+        streamDataWatcherThread.start();
+
+        /* waiting for process */
+        LOG.debug("Wait for GIT wrapper to finish process.");
+
+        boolean exitDoneInTime = false;
+        try {
+            exitDoneInTime = process.waitFor(minutesToWaitForResult, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("GIT wrapper could not finish process.", e);
+        }
+
+        if (!exitDoneInTime) {
+            LOG.error("GIT wrapper could not finish process. Waited {} minutes.", minutesToWaitForResult);
+            throw new RuntimeException("GIT wrapper could not finish process. Waited " + minutesToWaitForResult + " minutes.");
+        }
+        streamDatawatcherRunnable.stop();
+        checkProcessExitValue(process.exitValue());
+    }
+
+    private void checkProcessExitValue(int exitValue) {
+        String result = "";
+        if (exitValue == 0) {
+            result = "GIT wrapper finished process successfully.";
+        } else {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getProcess().getErrorStream()))) {
+                StringBuilder stringBuilder = new StringBuilder();
+                String line = null;
+                while (true) {
+                    try {
+                        if ((line = reader.readLine()) == null)
+                            break;
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                    stringBuilder.append(line);
+                    stringBuilder.append(System.getProperty("line.separator"));
+                }
+                result = stringBuilder.toString();
+            } catch (IOException e) {
+                throw new RuntimeException(result, e);
+            }
+            throw new RuntimeException(result);
+        }
+        LOG.debug(result);
+    }
+
+    private static class StreamDataRefreshRequestWatcherRunnable implements Runnable {
```

Comment:
IMO this runnable and the complete thread setup is currently doing nothing here but only wasting 300 milliseconds until being stopped - but I can see no other purpose.

IMO this could be comletely get rid off (at least in the current situation)